### DISCUSSION TITLE
Wiring Raspberry Pi 4B with Adafruit Feather RP2040 Scorpio

### DISCUSSION THREAD

**Discussion by delavandaag** [awawa-dev/HyperSPI#56] [Q&A] [answered]
What a great project! I very much want to build it for my TV, with a RPI 4B, Adafruit Feather RP2040 Scorpio for multi-segment and HyperSPI, a Mean Well 5V 15A power source, Freintech splitter, SK6812 WRGB (60 leds per meter) and an MS2130 grabber. 

As I could not find a schema that shows all physical connections I drafted one myself. 

As to that schema, I have three questions:

1. is it correct (and if not, what is wrong)?
2. as from what number of consecutive leds is it wise to power both ends of the led strips?
3. is there anything wrong with powering the HDMI splitter with the Mean Well power source, instead of with the RPI power source (as it is drawn in the schema)? It would save me having to split the USB cord of the RPI.

Best, Dela
![HyperSPI schema](https://github.com/user-attachments/assets/e962a9d9-bede-4424-bc4d-2bc4f0bb987d)

**Comment by awawa-dev** [comment]
Hi
Thank you! The diagram looks OK at first glance, except that I would route GND from RPi directly to GND rp2040, without any nodes in the way. There is also no point in connecting an additional +5V if you do not also connect GND (I mean the right side of these LEDs).
With 300 RGBW LEDs and a single setup, connecting at the beginning and end is enough, so in the case of a 2x150 segment, it should be enough to supply power only at one point on each segment. I don't know how this scales for a larger number of LEDs, you would have to measure the voltage at the furthest point from the power source on the segment.
I don't know how the splitter will behave with the slightly unstable behavior of the LED driver. I think that since it doesn't need much power it shouldn't cause problems so you can try it and see how it behaves.

As for Scorpio power supply:
-  Battery connector is for 3.7/4.2V LiPo battery
- the manufacturer note:
```

The Feather is not designed for external power supplies - this is a design decision to make the board compact and low cost. It is not recommended, but technically possible:

- Connect an external 3.3V power supply to the 3V and GND pins. Not recommended, this may cause unexpected behavior and the EN pin will no longer work. Also this doesn't provide power on BAT or USB and some Feathers/Wings use those pins for high current usages. You may end up damaging your Feather.
- Connect an external 5V power supply to the USB and GND pins. Not recommended, this may cause unexpected behavior when plugging in the USB port because you will be back-powering the USB port, which could confuse or damage your computer.

```

**Comment by delavandaag** [comment]
Many thanks, Awawa! Thanks to you it is now clear to me how to ground the rp2040. But it is unclear to me how to power the rp2040. Should I connect the 3.3V from the RPI (connection 1, top left in the schema) to one of the 3.3V pins on the rp20240 (second or third from the top left)?

**Reply by awawa-dev** [reply]
it wont work:
```
The Raspberry Pi 3.3V supply was designed with a maximum current of ~3mA per GPIO pin. If you load each pin with 16mA, the total current is 272mA. The 3.3V supply will collapse under that level of load. 
```
you can try 5v from the led power supply using alternative pins (USB and GND pins) remembering the manufacturer warning. Or use USB power supply.

**Comment by delavandaag** [comment]
Ah, thanks. I think I will order a cable with one USB C female connector and two USB C male connectors for that purpose, i.e. to connect both the RPI and the rp2040.

**Reply by delavandaag** [reply]
EDIT: (...) i.e. to connect both the RPI and the rp2040 to the RPI power supply.

**Reply by awawa-dev** [reply]
And why not connect rp2040 to RPI USB port?

**Reply by delavandaag** [reply]
Yeah, I did not know that the RPI USB port delivers 5V. Which is, I guess, a bit of a dumb assumption.

I think I took enough of your time, at least for this moment. I will redraft and publish my schema somewhere in the next week so that others can use it as an example. 

And if I face any problem on the software side of HyperSPI, I would appreciate your help for that. And I am sure you will jump in again then. Not to claim your help, but from what I see in this discussion thread you jump on every question. And that, Awawa, is amazing. Thanks, many thanks for that. And for your software and all improvements thereof!

**Reply by delavandaag** [reply]
Below is my redrafted schema. Except for the Feintech splitter, of which I received a non-working one, everything is connected as in the schema and it seems to work. I have not yet attached the LED strips to my TV as I have not yet had time for that, but I expect to find some time in the coming week.

![HyperSPI schema v4](https://github.com/user-attachments/assets/cedf5684-705e-44ca-b33d-2a18b6318da6)

**Reply by delavandaag** [reply]
Everything is installed and works brilliantly! As the TV is standing in front of a green wall I still have to find out how to best adjust the color settings, but I am very happy with the result. Oh, and I did not power the LED strips on both sides. Otherwise I connected according to my last schema.

Awawa, you write that the cables (including ground) should not exceed 15-20cm or it may be necessary to lower the SPI speed. I take it that this only refers to the four cables between the RPI and the Adafruit Feather RP2040 Scorpio and not to the connections between the Adafruit Feather RP2040 Scorpio and the LED strips. Is that correct?

**Reply by awawa-dev** [reply, marked as answer]
Great that it works!
> Oh, and I did not power the LED strips on both sides. Otherwise I connected according to my last scheme.

Yes, usually single segments at least up to 150 do not require double power. You can always add them later if needed.

> Awawa, you write that the cables (including ground) should not exceed 15-20cm or it may be necessary to lower the SPI speed. I take it that this only refers to the four cables between the RPI and the Adafruit Feather RP2040 Scorpio and not to the connections between the Adafruit Feather RP2040 Scorpio and the LED strips. Is that correct?

Yes, the main limitation is the SPI connection between the Rpi and Adafruit. The LED data connection, if a 3.3V to 5V level shifter is used (built into Adafruit), should work for at least 1 meter although this depends on the EMI environment (based on user reports mainly from the WLED project). If a level shifter is not used, the connection should also be as short as possible for such a setup to have a chance work at all.
